openweather-based-bidding/main.gs

```diff
@@ -35,86 +38,92 @@ function monitorWeatherAndSyncWithDV360() {
-    let tempConditionSatisfied = true;
-    if (tempMin || tempMin !== '') {
-      tempConditionSatisfied = (tempMin <= currentTemperature);
+    if (tempMin) {
+      satisfiedConditions.tempMin = tempMin <= row[config.get('col-temp-curr')];
     }
 
     // Check if max temperature condition is satisfied
-    if (tempMax || tempMax !== '') {
-      tempConditionSatisfied = tempConditionSatisfied
-        && (currentTemperature <= tempMax);
+    if (tempMax) {
+      satisfiedConditions.tempMax = row[config.get('col-temp-curr')] <= tempMax;
     }
 
     // Check if rain/snow condition is satisfied
-    let rainSnowConditionSatisfied = true;
     if (onlyIfRainingOrSnow) {
-      rainSnowConditionSatisfied = currentRainSnowCondition;
+      satisfiedConditions.precip = row[config.get('col-precip-curr')];
     }
 
     // Check if wind speed condition is satisfied
-    let windConditionSatisfied = true;
     if (windMin) {
-      windConditionSatisfied = (currentWindspeed >= windMin);
+      satisfiedConditions.windSpeed = row[config.get('col-wind-curr')] >= windMin;
     }
 
     // Check if all conditions are satisfied
-    const activate = rainSnowConditionSatisfied &&
-                      tempConditionSatisfied &&
-                      windConditionSatisfied;
-
-    // Log status to debug panel
-    Logger.log(
-      `Updating: IO: ${insertionOrderId}, LI: ${lineItemId}, `
-      + `Active: ${activate}, `
-      + `Temp Min.: ${tempMin}, Temp Max.: ${tempMax}, `
-      + `Curr. Temp.: ${currentTemperature}, `
-      + `Curr. Rain/Snow: ${currentRainSnowCondition}, `
-      + `Wind Min: ${windMin}, Curr. Wind speed: ${currentWindspeed}`
-    );
-
-    /*
-     * This does the API request to change the LI status.
-     * If you wish to change the IO status instead just change to:
-     * dv360.switchIOStatus(advertiserId, insertionOrderId, activate)
-     */
+    const activate = allObjectPropertiesTrue(satisfiedConditions);
+
+    // Switch Line Item status
     dv360.switchLIStatus(advertiserId, lineItemId, activate);
 
-    // Log the status to the spreadsheet
-    const row = i + 1;
-    sheet.getRange( 'J' + row ).setValue( currentTemperature );
-    sheet.getRange( 'K' + row ).setValue( currentRainSnowCondition );
-    sheet.getRange( 'L' + row ).setValue( currentWindspeed );
-    sheet.getRange( 'M' + row ).setValue( activate ? 'Active' : 'Paused' );
-    sheet.getRange( 'N' + row ).setValue( new Date().toISOString() );
+    // Switch Insertion Order status
+    dv360.switchIOStatus(advertiserId, insertionOrderId, activate)
+
+    // Set new status
+    row[config.get('col-status')] = activate ? 'Active' : 'Paused';
+
+    // Set last update timestamp
+    row[config.get('col-last-update')] = new Date().toISOString();
+
+    // Write back to Sheet
+    if (!sheetsApi.write([row], configSpreadsheetName + '!A' + (i + 1))) {
+      Logger.log('An error occurred, retrying in 30s');
+      Utilities.sleep(30000);
```

Comment:
1. I don't see how you retry in 30 secs (does not look like automatic sheets API functionality). 
2. If the row `i` has an error, then the next row `i+1` result will be written to the `i` row (since `i--`). It creates confusion and removes rows from the sheet
3. Too many 30 sec delays might exceed the script run time limit

I would suggest to try instead to write an error message to the row `i` with the `sheet.getRange( ... ).setValue('Error ...')`